tests/FsConfig.Tests/AppConfig.fs

```diff
@@ -67,6 +67,35 @@ module ``App Config tests`` =
 
             test <@ appConfig.Get<Config>() = expected @>
 
+        [<Test>]
+        member __.``getOptionalRecord should succeed``() =
+            let expected =
+                {
+                    optionalSubsectionNone = None
+                    optionalSubsectionSome = Some({ apiKey = "abc-def"; value = 1 })
+
+                }
+
+                |> Ok
+
+            test <@ appConfig.Get<ConfigWithOptionalSubsection>() = expected @>
+
+        // [<Test>] //TODO
```

Comment:
@queil is it something that you planned to add later?

Reply:
@tamizhvendan yes, support for lists of records. I'll remove it from here as the implementation is not as straightforward as I thought.

Reply:
I've just removed the work in progress bits.

Reply:
Yeah, It's a bit tricky. Thanks for the contribution :)